app/validators/embargo_date_parts.rb

```diff
@@ -0,0 +1,14 @@
+# typed: true
+# frozen_string_literal: true
+
+# Validates embargo month and day
+class EmbargoDateParts < ActiveModel::Validator
+  def validate(record)
+    return if record.send('embargo_date').instance_of?(Date)
+    return if record.send('release') == 'immediate'
```

Comment:
Can you move this up into the form?  E.g. `validates_with EmbargoDateParts, if: Proc.new { |form| form.release == 'immediate' }`

Reply:
Moving this up to the draft_work_form.rb causes four failing tests in the `spec/requests/create_work_spec.rb` (test on lines 328, 358, 396, 463).

Reply:
sorry, the `if` should be `unless`.  Does that fix it?

Reply:
Yup! That fixed the issues with the failing tests. Just pushed up a new commit.